### ISSUE TITLE
Locking: Checkboxes are off by default

### ISSUE THREAD

**Issue by jameskoster** [WordPress/gutenberg#40354]
When you select 'Lock' from the block toolbar, the resulting modal displays all the lock options toggled off by default:

https://user-images.githubusercontent.com/846565/163415139-ca1f7556-0310-4ec9-89a2-db1ee2a855f7.mp4

Since the action that invokes the modal is affirmative, shouldn't the checkboxes be _on_ by default? This would eliminate a click in the locking process when you want to fully lock a block down.

**Comment by apeatling** [comment]
Looking at this a bit more I think there's slightly more detail:

1. When selecting "Lock" from the menu, check all the boxes in the modal
2. When a block is locked or partially locked and you select the lock icon from the block toolbar, reflect the current state in the checkboxes
3. When the block is locked or partially locked and "Unlock" is selected from the menu, uncheck all the boxes in the modal

This should remove friction in both directions, but also preserve the current state if the lock icon I selected (which does not have an inherent action). Does this make sense, or is it adding a level of confusion around checkbox state and locked state?

**Comment by jameskoster** [comment]
> When a block is locked or partially locked and you select the lock icon from the block toolbar, reflect the current state in the checkboxes

This one already happens, no?

A second thought on this one, would it be strange if the modal _didn't_ reflect the current lock status? From that perspective it may be that the issue isn't how the options are presented, but rather the "Lock" and "Unlock" labels in the first place, which kind of suggest one action or the other. Perhaps if that label simply read "Lock / Unlock" that would be less ambiguous 🤔

**Comment by apeatling** [comment]
> This one already happens, no?

Yes, that would remain the same I wanted to make sure I had all the scenarios down. 😂

> A second thought on this one, would it be strange if the modal didn't reflect the current lock status?

I think this is my concern although I was expressing it indirectly -- you'd essentially be representing state using the checkboxes differently depending on the initial action.

"Lock / Unlock" would slightly improve the flow, but it's still three clicks to unlock or lock a block from the menu, and requires some thought when using it for the first time. I don't have any good other suggestions, but I'll think more.
